core/src/com/unciv/models/ruleset/Building.kt

```diff
@@ -247,6 +247,12 @@ class Building : NamedStats(), IConstruction {
                         && !construction.containsBuildingOrEquivalent(it.params[0])} )
             return "Should not be displayed"
 
+        if(uniqueObjects.any { it.placeholderText=="Not displayed as an available construction without []"
+                        && ((it.params[0] in civInfo.gameInfo.ruleSet.buildings &&!construction.containsBuildingOrEquivalent(it.params[0]))
```

Comment:
A. Add a val for `it.params[0]`, call it "filter" or something
B. Remove unnecessary parentheses, if you're using Android Studio it'll recommend this itself

Reply:
Actually now that I think of it there shouldn't be a reason to have the code for the units and buildings different at all, no?

Reply:
no, it makes sense that they are the same :)

Reply:
android studio complains if i remove any ( ), which ones are unnecessary?

Reply:
Never mind, it's not that important ;)